### DISCUSSION TITLE
Global Hotkeys

### DISCUSSION THREAD

**Discussion by TheRealDadbeard** [Chatterino/chatterino2#3673] [Ideas]
Right now you can create custom hotkeys etc. but non of them are global making them kinda useless, at least for me.

Having the ability to set my custom hotkeys to Global would be great as of right now I still have to have chatty run in the background for my hotkeys.

**Comment by Felanbird** [comment]
~~I'm confused by what you mean saying the hotkeys are "not global" - if the hotkeys were not global, they would only apply to the window you had focused when you created them, and wouldn't work in any other tab.~~

It's been noted to me by "global" you likely mean keybinds that work while Chatterino is not focused, but I'm unable to think of any useful examples excluding clipping `alt+x`

**Reply by LosFarmosCTL** [reply]
EDIT: whoops missed your edit 

I think he means that hotkeys are not picked up when chatterino is not focused (so global in terms of the OS?), but there is definitely no support for that in qt itself, and as far as i can tell after a bit of googling all libraries that existed through C bindings are discontinued.

So very, very unlikely that support for this will come at any point.

**Reply by jitspoe** [reply]
Why does it have to be supported by QT?  It would probably require a specific implementation per OS that taps into their native API's, but I don't see why the UI framework would limit that.

**Reply by TheRealDadbeard** [reply]
> 

To post commands in the chat while in-game. Like starting ads or triggering my own bot commands. Global hotkeys are extremely useful. 

I'm really confused how you can't see that.

**Reply by Mm2PL** [reply]
> Why does it have to be supported by QT? It would probably require a specific implementation per OS that taps into their native API's, but I don't see why the UI framework would limit that.

I'm sure you are **totally burning** to implement global hotkeys for Windows, MacOS and twice for Linux (X11, Wayland). I wouldn't want either, I believe that it's way more work than the hotkey system took and that one took me a good half a year to finalize to a stable state. The only thing we might get away with ignoring is Wayland, because XWayland exists and global hotkeys might work with it. There exist projects that aim to provide global hotkey functionality, however we would probably need to update them when we move to Qt6.

**Reply by Felanbird** [reply]
> I'm really confused how you can't see that.

Because you made no mention of what functionality you'd actually wanted, if you have a specific use-case for your request you should mention it, it makes your argument better.

**Reply by TheRealDadbeard** [reply]
Well I said it would be nice to have them global which I thought would imply them being systemwide without Chatterino being in focus. They wouldn't be global if chatterino has to be in focus.

**Reply by LosFarmosCTL** [reply]
https://github.com/Skycoder42/QHotkey

Just dropping this here for reference if someone would actually want to investigate this, I have not used that library and basically didn't look into it further than the README, but apparently it at least works for Windows, macOS, X11 and would support Qt 5 and 6

**Reply by leon-richardt** [reply]
> The only thing we might get away with ignoring is Wayland, because XWayland exists and global hotkeys might work with it.

"Dread it, run from it, Wayland arrives all the same."
